src/VideoManager/SubtitleWriter.cc

```diff
@@ -101,22 +125,31 @@ void SubtitleWriter::_captureTelemetry()
     QStringList valuesStrings;
 
     // Make a list of "factname:" strings and other with the values, so one can be aligned left and the other right
-    for (const auto& i : _values) {
-        valuesStrings << QStringLiteral("%2 %3").arg(vehicle->getFact(i)->cookedValueString())
-                                                .arg(vehicle->getFact(i)->cookedUnits());
-        namesStrings << QStringLiteral("%1:").arg(vehicle->getFact(i)->shortDescription());
+    for (const FactPath& fact : _facts) {
+        //Vehicle facts
+        if(fact.groupName == vehicleFactGroupName || fact.groupName == ""){
+            valuesStrings << QStringLiteral("%2 %3").arg(vehicle->getFact(fact.factName)->cookedValueString())
```

Comment:
Maybe having a helper variable for `auto factData = vehicle->getFact(fact.factName)` to help with the readability of the code ?
You can also change this variable for groupName and remove all the code inside the if/else to be common and only changing the factData